### ISSUE TITLE
Better UX for recursive questions

### ISSUE THREAD

**Issue by sekoyo** [SBoudrias/Inquirer.js#595]
Hi, great lib works nicely. I'm trying to get the user to select a path in the current directory. I have a recursive prompt like this:

```js
function choosePathFrom(currentPath, cb) {
  const directories = getDirectories(currentPath)
    .filter(directory => ignoredDirectories.indexOf(directory) === -1);

  directories.push(new inquirer.Separator());
  directories.push({
    name: `Choose this path ${path.relative(process.cwd(), currentPath)}`,
    value: null,
  });

  pathQuestion[0].choices = directories;

  inquirer.prompt(pathQuestion).then((answers) => {
    if (answers.path === null) {
      cb(currentPath);
    } else {
      choosePathFrom(path.join(currentPath, answers.path), cb);
    }
  });
}

function promptForLocalePath() {
  const currentPath = path.resolve();
  return new Promise(resolve => choosePathFrom(currentPath, resolve))
    .then((localePath) => {
      console.log('Finished selecting:', localePath);
    });
}
```

It works well but I'm not really satisfied with the output as you're choosing the directory:

![recursive-path](https://user-images.githubusercontent.com/760314/31196018-1594dec8-a944-11e7-98b7-71c991f880b7.gif)

Is there a way to "swallow" up the output as they are choosing a directory? Thanks

**Comment by rayk** [comment]
The repeating of the questions is a problem, I found it mentioned in https://github.com/SBoudrias/Inquirer.js/issues/432 also. 

I have noticed the problem does not occur if you have a function per question, but using a recursive approach like above it is real problem.  Basically this bugs makes it harder to keep code DRY.

for example, a generic navigation function like the one below:
```js
const nav = session => {
    const menu = session.menuNext;
    out(chalk.green.bold(menu.name));
    inquire.prompt(menu).then(choice => {
        R.head(R.values(choice))(R.pipe(updateHistory, setPreviousMenu, clearNextMenu)(session));
    });
};
```
always cause the question to be repeated.
